src/zenml/materializers/built_in_materializer.py

```diff
@@ -169,14 +169,48 @@ def find_type_by_str(type_str: str) -> Type[Any]:
     Returns:
         The type whose string representation is `type_str`.
     """
-    # TODO: how to handle subclasses of registered types?
     registered_types = default_materializer_registry.materializer_types.keys()
     type_str_mapping = {str(type_): type_ for type_ in registered_types}
     if type_str in type_str_mapping:
         return type_str_mapping[type_str]
     raise RuntimeError(f"Cannot resolve type '{type_str}'.")
 
 
+def find_materializer_registry_type(type_: Type[Any]) -> Type[Any]:
```

Comment:
This seems like it would be a useful method in the `MaterializerRegistry` instead?